packages/beacon-node/src/network/peers/discover.ts

```diff
@@ -131,17 +133,31 @@ export class PeerDiscovery {
   async start(): Promise<void> {
     await this.discv5.start();
     this.discv5StartMs = Date.now();
-    this.discv5.on("discovered", this.onDiscovered);
+
+    for (const service of ((this.libp2p as unknown) as {services: unknown[]}).services) {
+      if (isPeerDiscovery(service)) {
+        service.addEventListener("peer", this.onDiscoveredPeer);
+      }
+    }
+
+    this.discv5.on("discovered", this.onDiscoveredENR);
     if (this.connectToDiscv5BootnodesOnStart) {
       // In devnet scenarios, especially, we want more control over which peers we connect to.
       // Only dial the discv5.bootEnrs if the option
       // network.connectToDiscv5Bootnodes has been set to true.
-      this.discv5.kadValues().forEach((enr) => this.onDiscovered(enr));
+      this.discv5.kadValues().forEach((enr) => this.onDiscoveredENR(enr));
     }
   }
 
   async stop(): Promise<void> {
-    this.discv5.off("discovered", this.onDiscovered);
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+    for (const service of (this.libp2p as any).services) {
+      if (isPeerDiscovery(service)) {
+        service.removeEventListener("peer", this.onDiscoveredPeer);
```

Comment:
Why don't we bake mdns into peer discovery instead of having to add it to libp2p?

Reply:
Ended up going the other direction. Listening to libp2p's "peer:discovery" event